File: app/src/main/java/maderski/chargingindicator/Receivers/BatteryReceiver.java
package maderski.chargingindicator.Receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import maderski.chargingindicator.Battery.BatteryManager;
import maderski.chargingindicator.Notification.NotificationManager;
import maderski.chargingindicator.Actions.PerformActions;
import maderski.chargingindicator.Services.BatteryService;

/**
 * Created by Jason on 8/9/16.
 */
public class BatteryReceiver extends BroadcastReceiver {
    private static final String TAG = "BatteryReceiver";

    private BatteryManager mBatteryManager;

    public BatteryReceiver(){}

    public BatteryReceiver(BatteryManager batteryManager) { mBatteryManager = batteryManager;}

    @Override
    public void onReceive(Context context, Intent intent) {
        mBatteryManager.setBatteryStatus(intent);
        if(mBatteryManager.isPluggedIn()) {
            PerformActions performActions = new PerformActions(context);
            performActions.makeBatteryChargedSound(context, performActions, mBatteryManager);

            NotificationManager notificationManager = new NotificationManager(context, mBatteryManager);
            notificationManager.setNotifMessage();
        }
    }
}

File: app/src/main/java/maderski/chargingindicator/receivers/BatteryReceiver.kt
package maderski.chargingindicator.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import maderski.chargingindicator.CIApplication
import maderski.chargingindicator.R
import maderski.chargingindicator.actions.interfaces.PerformActions
import maderski.chargingindicator.helpers.battery.BatteryHelper
import maderski.chargingindicator.utils.ServiceUtils
import javax.inject.Inject

class BatteryReceiver : BroadcastReceiver() {
    @Inject
    lateinit var performActions: PerformActions

    @Inject
    lateinit var batteryHelper: BatteryHelper

    private var canChargedSoundPlay = true

    init {
        CIApplication.instance.appComponent.inject(this)
    }

    override fun onReceive(context: Context?, intent: Intent?) {
        val appContext = context?.applicationContext
        if (appContext != null && intent != null) {
            val pendingResult: PendingResult = goAsync()
            val isBatteryCharged = batteryHelper.isBatteryUserCharged(appContext, intent)
            val title = "Battery Level: ${batteryHelper.batteryLevel(intent)}"
            val message = if (isBatteryCharged) "CHARGED!" else "Charging..."
            val icon = if (isBatteryCharged) android.R.drawable.ic_lock_idle_charging else R.drawable.standardbolt
            Log.d(TAG, title)
            playBatteryChargedSound(isBatteryCharged)
            updateNotification(appContext, title, message, icon)
            pendingResult.finish()
        }
    }

    private fun playBatteryChargedSound(isBatteryCharged: Boolean) {
        if (isBatteryCharged && canChargedSoundPlay) {
            performActions.batteryChargedSound()
            canChargedSoundPlay = false
        }
    }

    private fun updateNotification(
        appContext: Context,
        title: String,
        message: String,
        icon: Int
    ) {
        ServiceUtils.updateServiceNotification(
            ServiceUtils.FOREGROUND_NOTIFICATION_ID,
            title,
            message,
            appContext,
            appContext.getString(R.string.ci_channel_id),
            appContext.getString(R.string.ci_channel_name),
            icon,
            true
        )
    }

    companion object {
        const val TAG = "BatteryReceiver"
    }
}